libopenage/input/action.cpp

```diff
@@ -3,18 +3,44 @@
 #include <functional>
 
 #include "action.h"
+#include "../engine.h"
 
 namespace openage {
 namespace input {
 
+bool ActionManager::create(const std::string type) {
+	if (actions.find(type) == actions.end()) {
+		actions.insert(std::make_pair(type, actions.size()));
+		return true;
+	}
+	return false;
+}
+
+action_t ActionManager::get(const std::string type) {
+	auto it = actions.find(type);
+	if ( it != actions.end()) {
+		return it->second;
+	}
+	return actions.at("UNDEFINED");
```

Comment:
can you please add `this->` for all the member accesses? (there's a space too much in front of `it` btw)